crates/metadata-store/src/local/tests.rs

```diff
@@ -362,8 +366,16 @@ async fn start_metadata_store(
     )?;
 
     // await start-up of metadata store
+    let metadata_store_address = if let config::MetadataStore::Embedded { address } =
+        metadata_store_client_options.metadata_store_client
+    {
+        address.clone()
+    } else {
+        unimplemented!()
+    };
```

Comment:
For cases where you expect a single variant, you can use `let_assert!(config::MetadataStore::Embedded { address } = metadata_store_client_options.metadata_store_client)` to destructure the type.